clang/lib/AST/ExprConcepts.cpp

```diff
@@ -153,6 +153,10 @@ RequiresExpr::RequiresExpr(ASTContext &C, SourceLocation RequiresKWLoc,
   std::copy(Requirements.begin(), Requirements.end(),
             getTrailingObjects<concepts::Requirement *>());
   RequiresExprBits.IsSatisfied |= Dependent;
+  RequiresExprBits.IsSatisfied &=
+      llvm::none_of(LocalParameters, [](const ParmVarDecl *Param) {
+        return Param->isInvalidDecl();
+      });
```

Comment:
I don't understand why we need to say the requires expression is not satisfied if any of the parameters are invalid. This is the consequence of an error, which was already emitted.

This does not seem to lead to better error recovery.

Reply:
I asked author to add it because I saw GCC evaluating such expressions as false.

https://gcc.godbolt.org/z/MYz1oa1TT

This is seemingly more of an open question IMO, given that msvc & edg don't agree with gcc.

Reply:
Yeah sure GCC does it, but IMO it looks worse as it results in a double error.

Reply:
OK I'm sold - now I lean towards your idea of not marking them invalid :)